src/components/JobManagementTable.tsx

```diff
@@ -0,0 +1,205 @@
+'use client';
+import React, { useEffect, useState } from 'react';
+import {
+  Table,
+  TableBody,
+  TableCell,
+  TableHead,
+  TableHeader,
+  TableRow,
+} from './ui/table';
+import { DeleteIcon, Edit } from 'lucide-react';
+import { getAllJobsAdditonalType, JobType } from '@/types/jobs.types';
+import { ServerActionReturnType } from '@/types/api.types';
+import { useToast } from './ui/use-toast';
+import {
+  Dialog,
+  DialogContent,
+  DialogDescription,
+  DialogFooter,
+  DialogHeader,
+  DialogTitle,
+  DialogTrigger,
+} from '@/components/ui/dialog';
+import { Button } from './ui/button';
+import { JobQuerySchemaType } from '@/lib/validators/jobs.validator';
+import { DEFAULT_PAGE, JOBS_PER_PAGE } from '@/config/app.config';
+import { Pagination, PaginationContent, PaginationItem } from './ui/pagination';
+import {
+  PaginationNextButton,
+  PaginationPreviousButton,
+} from './pagination-client';
+import APP_PATHS from '@/config/path.config';
+import { PaginationPages } from './ui/paginator';
+import {
+  approveJob,
+  deleteJobById,
+  getAllJobsForAdmin,
+} from '@/actions/job.action';
+import ApproveJobDialog from './ApproveJobDialog';
+
+type props = {
+  searchParams: JobQuerySchemaType;
+  jobs: ServerActionReturnType<getAllJobsAdditonalType>;
+  initialJobs: JobType[];
+};
+
+const JobManagementTable = ({ jobs, searchParams, initialJobs }: props) => {
+  const { toast } = useToast();
+  const [DispJobs, setDispJobs] = useState<JobType[]>(initialJobs);
```

Comment:
instead of just the initial jobs , can you make all of them server rendered ?

Reply:
it was server rendered before, but because of server renders when something gets change like if job approves then the datatable was not syncing with the DB unless u hard refresh

Reply:
you'll have to revalidate the route , and then it'll work